### PR TITLE
Added price domain object structure

### PR DESCRIPTION
**Pull request by tonasanya**

### CODE REVIEW

File: src/domain_factory/price/discount_rate_factory.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from domain_factory.price.price_factory import PriceFactory
+
+
+class InterestRateFactory(PriceFactory):
```
Comment (GalateaTom) [inline src/domain_factory/price/discount_rate_factory.py]: I think this class value and the comments needs to be updated from Interest to Discount

File: src/domain_factory/product/product_factory.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def generate_product_id(self, id):
             A single product Id
         """
 
-        return 1000 + id
+        return id
```
Comment (GalateaTom) [inline src/domain_factory/product/product_factory.py]: Why have you removed the 1000 in the front?  If you think that there shouldn't be numbers in front of IDs should you raise a fresh issue and make the change in product_factory.py and brokerage_account_factory.py

**Review by GalateaTom — CHANGES_REQUESTED** [review]
See comments

File: src/domain_factory/price/discount_rate_factory.py (hunk 1 above)
Comment (tonasanya) [reply src/domain_factory/price/discount_rate_factory.py]: done

File: src/domain_factory/product/product_factory.py (hunk 2 above)
Comment (tonasanya) [reply src/domain_factory/product/product_factory.py]: added back. As discussed, any dependant will retrieve created productIds from the database

**Review by GalateaTom — APPROVED** [review]

**Pull request merged**